internal/diskcache/cache.go

```diff
@@ -126,7 +126,7 @@ func (s *Store) OpenWithPath(ctx context.Context, key []string, fetcher FetcherW
 	go func(ctx context.Context) {
 		if s.BackgroundTimeout != 0 {
 			var cancel context.CancelFunc
-			ctx, cancel = context.WithTimeout(context.Background(), s.BackgroundTimeout)
+			ctx, cancel = WithIsolatedTimeout(ctx, s.BackgroundTimeout)
```

Comment:
It looks like this child context doesn't reference the parent's timeout, so isn't the right thing to do here to create a new context from the parent?

```suggestion
			ctx, cancel = context.WithTimeout(ctx, s.BackgroundTimeout)
```

The error before was the use of `context.Background` rather than the provided `ctx context.Context`

Reply:
Doing this should preserve all the values of `ctx`, with the given timeout

Reply:
Thats exactly what we dont want to do. The parents timeout may be shorter than the background timeout, and we dont want to kill the background task before s.BackgroundTimeout

Reply:
any parent timeout cascades downwards, which we dont want to happen. The background task should continue until s.BackgroundTimeout even if the parent context has been cancelled/timedout

Reply:
Ah I see! Thanks for the clarification 😁

Reply:
Had to double check coz I was doubting myself now :laughing: but yea that is the case
```go
	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
	defer cancel()
	ctx, cancel = context.WithTimeout(ctx, time.Minute)
	defer cancel()
	<-ctx.Done()
```